Alertmanager: Advanced configuration fields

dtaylor113 opened the pull request:
Added advanced configuration fields within expand/collapse section on each receiver form.

### [Screenshots](https://docs.google.com/document/d/1XweYUBmfYeUEVir1ajPSvrdwBBZS0awct_i0_M9bDOY/edit?usp=sharing)

dtaylor113 commented:
Hi @benjaminapetersen, @kyoto, and @spadgett,  I believe this PR is ready to be reviewed.

benjaminapetersen commented:
/hold

We need to know if the API is going to change on this, and if monitoring is working on that at this point.  If it changes, we may want to revise to the new API rather than merging right now.

spadgett commented:
/hold cancel

spadgett commented on frontend/integration-tests/views/monitoring.view.ts:
```diff
@@ -65,7 +65,8 @@ export const getFirstRowAsText = () => {
 export const saveAsDefault = firstElementByTestID('save-as-default');
-
+export const sendResolvedAlerts = firstElementByTestID('send-resolved-alerts');
+export const showAdvConfiguration = $('button.pf-c-expandable__toggle');
```
nit: Let's spell out the word. It's already a long name and editors have autocomplete.

```suggestion
export const showAdvancedConfiguration = $('button.pf-c-expandable__toggle');
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx:
```diff
@@ -465,6 +482,27 @@ export const SaveAsDefaultCheckbox: React.FC<SaveAsDefaultCheckboxProps> = ({
 
+export const SendResolvedAlertsCheckbox = ({ formField, formValues, dispatchFormChange }) => {
+  return (
+    <label className="co-no-bold" htmlFor={formField}>
```
Either use the PF4 check or Bootstrap `checkbox` class instead of `co-no-bold`.

spadgett commented on frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx:
```diff
@@ -465,6 +482,27 @@ export const SaveAsDefaultCheckbox: React.FC<SaveAsDefaultCheckboxProps> = ({
+        checked={formValues[formField]}
+        aria-checked={formValues[formField]}
+      />
+      &nbsp; Send resolved alerts to this receiver?
```
`&nbsp;` shouldn't be needed if you use the checkbox component or Bootstrap checkbox classes.

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+              <label className="control-label" htmlFor="slack-icon-type">
+                Icon &nbsp;
+                <RadioInput
+                  title="Url"
```
```suggestion
                  title="URL"
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                    }
+                  />
+                  <div className="help-block" id="slack-icon_url-help">
+                    The url of the icon
```
All of the help blocks should have punctuation to be consistent with other forms in console (although it looks like existing help blocks for Alertmanager is missing that).

```suggestion
                    The URL of the icon.
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                      href="https://www.webfx.com/tools/emoji-cheat-sheet/"
+                      text="emoji code"
+                    />{' '}
+                    string to use in place of the default icon
```
```suggestion
                    to use in place of the default icon.
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                className="pf-c-form-control"
+                type="text"
+                aria-describedby="slack-username-help"
+                id="slackUsername"
```
IDs should be kebab case

```suggestion
                id="slack-username"
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                    type="text"
+                    aria-describedby="slack-icon_emoji-help"
+                    name="slackIconEmoji"
+                    data-test-id="slack-icon_emoji"
```
Use kebab case

```suggestion
                    data-test-id="slack-icon-emoji"
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+              )}
+              {formValues.slackIconType === 'emoji' && (
+                <>
+                  <input
```
Needs an `aria-label` if we don't have a `label` element.

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+              </label>
+              {formValues.slackIconType === 'url' && (
+                <>
+                  <input
```
Needs an `aria-label` if we don't have a `label` element.

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                }
+              />
+              <div className="help-block" id="slack-username-help">
+                The displayed username
```
```suggestion
                The displayed username.
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+              <label className="co-no-bold" htmlFor="slackLinkNames">
+                <input
+                  type="checkbox"
+                  id="slackLinkNames"
```
Kebab case

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+              </div>
+            </div>
+            <div className="form-group">
+              <label className="co-no-bold" htmlFor="slackLinkNames">
```
Avoid `co-no-bold` and use proper checkbox styles

spadgett commented on frontend/public/components/monitoring/receiver-forms/slack-receiver-form.tsx:
```diff
@@ -72,6 +85,144 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                &nbsp; Link Names
+              </label>
+              <div className="help-block" id="slack-link-names-help">
+                Find and link channel names and usernames
```
```suggestion
                Find and link channel names and usernames.
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/pagerduty-receiver-form.tsx:
```diff
@@ -113,6 +125,110 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                className="pf-c-form-control"
+                type="text"
+                aria-describedby="client-help"
+                id="pagerdutyClient"
```
Kebab case

spadgett commented on frontend/public/components/monitoring/receiver-forms/pagerduty-receiver-form.tsx:
```diff
@@ -113,6 +125,110 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                }
+              />
+              <div className="help-block" id="client-help">
+                The client identification of the Alertmanager
```
```suggestion
                The client identification of the Alertmanager.
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/pagerduty-receiver-form.tsx:
```diff
@@ -113,6 +125,110 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                className="pf-c-form-control"
+                type="text"
+                aria-describedby="client-url-help"
+                id="pagerdutyClientUrl"
```
kebab case

spadgett commented on frontend/public/components/monitoring/receiver-forms/pagerduty-receiver-form.tsx:
```diff
@@ -113,6 +125,110 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                }
+              />
+              <div className="help-block" id="client-url-help">
+                A backlink to the sender of the notification
```
```suggestion
                A backlink to the sender of the notification.
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/pagerduty-receiver-form.tsx:
```diff
@@ -113,6 +125,110 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                className="pf-c-form-control"
+                type="text"
+                aria-describedby="description-help"
+                id="pagerdutyDescription"
```
kebab case

spadgett commented on frontend/public/components/monitoring/receiver-forms/pagerduty-receiver-form.tsx:
```diff
@@ -113,6 +125,110 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                }
+              />
+              <div className="help-block" id="description-help">
+                Description of the incident
```
```suggestion
                Description of the incident.
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/pagerduty-receiver-form.tsx:
```diff
@@ -113,6 +125,110 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                className="pf-c-form-control"
+                type="text"
+                aria-describedby="severity-help"
+                id="pagerdutySeverity"
```
kebab case

spadgett commented on frontend/public/components/monitoring/receiver-forms/pagerduty-receiver-form.tsx:
```diff
@@ -113,6 +125,110 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                }
+              />
+              <div className="help-block" id="severity-help">
+                Severity of the incident
```
```suggestion
                Severity of the incident.
```

spadgett commented on frontend/public/components/monitoring/receiver-forms/email-receiver-form.tsx:
```diff
@@ -249,19 +245,55 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+                className="pf-c-form-control"
+                type="text"
+                aria-describedby="html-help"
+                id="emailHtml"
```
kebab case

spadgett commented on frontend/public/components/monitoring/receiver-forms/email-receiver-form.tsx:
```diff
@@ -249,19 +245,55 @@ export const Form: React.FC<FormProps> = ({ globals, formValues, dispatchFormCha
+            </div>
+            <div className="form-group">
+              <label className="control-label co-required" htmlFor="emailHtml">
+                Body of email notifications (HTML)
```
Use consistent case for all labels

```suggestion
                Body of Email Notifications (HTML)
```

dtaylor113 commented:
Hi @spadgett, I believe I have addressed all of your review comments.  Last [commit](https://github.com/openshift/console/pull/4044/commits/20ec3110dffb5ec9eddd90ddb19c29d83bea64a1) has just those changes. -thanks

spadgett commented:
/hold
for #5100

spadgett commented:
/hold cancel
/retest

dtaylor113 commented:
/retest
